DatadogFlags/Sources/Client/FlagAssignmentsFetcher.swift

```diff
@@ -74,6 +74,22 @@ internal final class FlagAssignmentsFetcher: FlagAssignmentsFetching {
                     case .success(let data):
                         do {
                             let response = try Self.decoder.decode(FlagAssignmentsResponse.self, from: data)
+
+                            // Log any flags that failed to decode to telemetry
+                            if !response.failedFlags.isEmpty {
+                                for (flagKey, errorDescription) in response.failedFlags {
+                                    let error = InternalError(description: errorDescription)
+                                    DD.logger.warn(
+                                        "Failed to decode flag '\(flagKey)' from flag assignments response. Flag will be dropped from configuration.",
+                                        error: error
+                                    )
+                                    featureScope.telemetry.error(
+                                        "Failed to decode flag '\(flagKey)' from flag assignments response",
+                                        error: error
+                                    )
```

Comment:
minor: I would use `debug` instead of error for the telemetry, since this is a situation we expect in the future.
```suggestion
                                    featureScope.telemetry.debug(
                                        "Failed to decode flag '\(flagKey)' from flag assignments response",
                                        error: error
                                    )
```